spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/JpaQueryMethod.java

```diff
@@ -347,11 +347,9 @@ public String getRequiredAnnotatedQuery() throws IllegalStateException {
 
 		String query = getAnnotatedQuery();
 
-		if (query != null) {
-			return query;
-		}
-
-		throw new IllegalStateException(String.format("No annotated query found for query method %s", getName()));
+		Assert.notNull(query, () -> String.format("No annotated query found for query method %s", getName()));
```

Comment:
This change changes behavior into throwing `IllegalArgumentException` instead of `IllegalStateException`. At that point, throwing an explicit `IllegalStateException` is fine so we should not change this method.